src/libstore/build/local-derivation-goal.cc

```diff
@@ -2522,6 +2529,11 @@ SingleDrvOutputs LocalDerivationGoal::registerOutputs()
                             wanted.to_string(SRI, true),
                             got.to_string(SRI, true)));
                 }
+                if (!newInfo.references.empty())
+                    delayedException = std::make_exception_ptr(
+                        BuildError("illegal path references in fixed-output derivation '%s'",
+                            worker.store.printStorePath(drvPath)));
```

Comment:
Hmm... I think this might be fixing a big? Or at least making a more explicit error message?

CA paths depend on the references, and fixed-output derivations have fixed CA output paths: therefore the references must be fixed ahead of time.

A previous version of this PR flirted with extending fixed-output derivations to support fixed non-trivial references, but I ripped that out to avoid any mission creep.

Reply:
This is a sensible restriction. If someone needs it removed, we can rely on them to start a conversation and we'll discover the best end state. Resolving.